lib/replicator.js

```diff
@@ -1834,10 +1845,18 @@ module.exports = class Replicator {
     if (b === null) return false
 
     removeInflight(b.inflight, req)
+    removeHotswap(b)
     b.queued = false
 
     b.resolve(value)
 
+    if (b.inflight.length > 0) {
```

Comment:
(detail) The if is not strictly needed here, the for does it implicitly

Reply:
true, let me add a comment, why not